tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,110 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExtendedFxConversionService;
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.math.RoundingMode;
+import java.util.stream.Collectors;
+
+public class ExtendedCurrencyCalculator extends CurrencyCalculator implements ExtendedFxConversionService {
+    private int rounding_mode;
+
+    public ExtendedCurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+    }
+
+    private Optional<BigDecimal> extendedOperation(ClientOperation operation, Beneficiary beneficiary) {
+        Optional<Quote> quote = findQuote(new CompareQuotesByPrice(operation, beneficiary), quotes);
+        if (!quote.isPresent()) {
+            System.out.println("Return Optional.empty()");
+            return Optional.empty();
+        }
+        showQuote(quote.get());
+        if (operation == ClientOperation.SELL) {
+                System.out.println("Returl value: " + BigDecimal.valueOf(1).divide(quote.get().getBid(),10, rounding_mode));
```

Comment:
Line is longer than 120 characters (found 124).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a854586832>*